src/services/refactors/inlineVariable.ts

```diff
@@ -0,0 +1,131 @@
+
+        if (refactor.isRefactorErrorInfo(info) && preferences.provideRefactorNotApplicableReason) {
+            return [{
+                name: refactorName,
+                description: refactorDescription,
+                actions: [{
+                    ...inlineVariableAction,
+                    notApplicableReason: info.error
+                }]
+            }];
+        }
+
+        if (info) {
+            return [{
+                name: refactorName,
+                description: refactorDescription,
+                actions: [inlineVariableAction]
+            }];
+        }
+
+        return emptyArray;
+    },
+
+    getEditsForAction(context, actionName) {
+        Debug.assert(actionName === refactorName, "Unexpected refactor invoked");
+
+        const {
+            cancellationToken,
+            file,
+            program,
+            startPosition
+        } = context;
+
+        const info = getInliningInfo(file, startPosition, program, cancellationToken);
+
+        if (!info || refactor.isRefactorErrorInfo(info)) {
+            return undefined;
+        }
+
+        const { references, declaration, replacement } = info;
+        const edits = textChanges.ChangeTracker.with(context, tracker => {
+            for (const node of references) {
+                tracker.replaceNode(file, node, replacement);
+            }
+            tracker.delete(file, declaration);
+        });
+
+        return { edits };
+    }
+});
+
+function getInliningInfo(file: SourceFile, startPosition: number, program: Program, cancellationToken: CancellationToken = NoopCancellationToken): InliningInfo | RefactorErrorInfo | undefined {
+    const token = getTokenAtPosition(file, startPosition);
+    const parent = token.parent;
+
+    // Make sure the token is inside a variable declaration and the declaration
+    // is not in a catch clause or for-loop.
+    if (isVariableDeclaration(parent) && isVariableDeclarationInVariableStatement(parent)) {
+        // We only care if the declaration has a value (since that's the thing we're inlining).
+        if (!parent.initializer) {
+            return undefined;
+        }
+
+        // Find all references to the variable.
+        const name = parent.name;
+        const referencedSymbols = FindAllReferences.Core.getReferencedSymbolsForNode(name.pos, name, program, program.getSourceFiles(), cancellationToken);
+        if (!referencedSymbols || referencedSymbols.length !== 1) {
+            return undefined;
+        }
+
+        // Only replace node references, and exclude the original variable too.
+        const references = mapDefined(referencedSymbols[0].references, entry =>
+            entry.kind === FindAllReferences.EntryKind.Node ? entry.node === name ? undefined : entry.node : undefined
+        );
+
+        return references.length === 0 ? undefined : { references, declaration: parent, replacement: parent.initializer };

```

Comment:
The problem with this is that if the formatter runs on this, it's possibly going to lose its mind if it encounters nodes that it's already formatted. I think you need to do a deep clone.

See if you can trigger an error through a test and then harden against that case.